lib/ViewportGizmo.ts

```diff
@@ -522,6 +541,18 @@ export class ViewportGizmo extends Object3D<ViewportGizmoEventMap> {
       .lookAt(camera.position, focusPoint, this.up);
     this._quaternionStart.setFromRotationMatrix(_matrix);
 
+    // For Z-up and X-up systems, when rotating to the top or bottom, we need to apply
+    // a final rotational twist to correctly align the gizmo's "Top" or "Bottom" face.
+    if (Object3D.DEFAULT_UP.z === 1 && Math.abs(position.z) > 0.99) {
+      const zSign = Math.sign(position.z);
+      this._targetQuaternion.multiply(zSign === 1 ? _negativeZQuaternion : _positiveZQuaternion);
+      this._quaternionEnd.multiply(zSign === 1 ? _negativeZQuaternion : _positiveZQuaternion);
+    } else if (Object3D.DEFAULT_UP.x === 1 && Math.abs(position.x) > 0.99) {
+      const xSign = Math.sign(position.x);
+      this._targetQuaternion.multiply(xSign === 1 ? _negativeZQuaternion : _positiveZQuaternion);
+      this._quaternionEnd.multiply(xSign === 1 ? _negativeZQuaternion : _positiveZQuaternion);
+    }
```

Comment:
This change applies a fix to normalize Z-up and Y-up camera animations from start to finish.